jetty-util/src/main/java/org/eclipse/jetty/util/thread/AutoLock.java

```diff
@@ -34,6 +34,38 @@
 
     private final ReentrantLock _lock = new ReentrantLock();
 
+    public static final AutoLock NO_LOCK = new AutoLock()
+    {
+        @Override
+        public AutoLock lock()
+        {
+            return this;
+        }
+
+        @Override
+        public boolean isHeldByCurrentThread()
+        {
+            return false;
+        }
+
+        @Override
+        public Condition newCondition()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        boolean isLocked()
+        {
+            return false;
+        }
+
+        @Override
+        public void close()
+        {
+        }
+    };
```

Comment:
@gregw this does not make sense.

This is only necessary because of a refactoring in the strategy, only to _try_ to simplify a perfectly readable algorithm into something that is now completely unreadable that probably also confuses the JIT.